File: sdkwork-ai-sdk-go/api/video.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type VideoApi struct {
    client *sdkhttp.Client
}

func NewVideoApi(client *sdkhttp.Client) *VideoApi {
    return &VideoApi{client: client}
}

// List videos
func (a *VideoApi) ListVideos(query map[string]interface{}) (sdktypes.OpenAiVideoList, error) {
    raw, err := a.client.Get(AiApiPath("/videos"), query, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideoList
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideoList](raw)
}

// Create video
func (a *VideoApi) CreateVideo(body sdktypes.VideoGenerationRequest) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Post(AiApiPath("/videos"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Create video
func (a *VideoApi) CreateVideoGenerations(body sdktypes.VideoGenerationRequest) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Post(AiApiPath("/videos/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Create video
func (a *VideoApi) CreateVideoGenerations2(body sdktypes.VideoGenerationRequest) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Post(AiApiPath("/v1/videos/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Retrieve video
func (a *VideoApi) Retrieve(video_id string) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/videos/%s", video_id)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Delete video
func (a *VideoApi) DeleteVideo(video_id string) (sdktypes.OpenAiVideoDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/videos/%s", video_id)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideoDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideoDeleteResponse](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/video.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { OpenAiVideo, OpenAiVideoDeleteResponse, OpenAiVideoList, VideoGenerationRequest } from '../types';


export class VideoApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** List videos */
  async listVideos(params?: QueryParams): Promise<OpenAiVideoList> {
    return this.client.get<OpenAiVideoList>(aiApiPath(`/videos`), params);
  }

/** Create video */
  async createVideo(body: VideoGenerationRequest): Promise<OpenAiVideo> {
    return this.client.post<OpenAiVideo>(aiApiPath(`/videos`), body);
  }

/** Create video */
  async createVideoGenerations(body: VideoGenerationRequest): Promise<OpenAiVideo> {
    return this.client.post<OpenAiVideo>(aiApiPath(`/videos/generations`), body);
  }

/** Create video */
  async createVideoGenerations2(body: VideoGenerationRequest): Promise<OpenAiVideo> {
    return this.client.post<OpenAiVideo>(aiApiPath(`/v1/videos/generations`), body);
  }

/** Retrieve video */
  async retrieve(video_id: string | number): Promise<OpenAiVideo> {
    return this.client.get<OpenAiVideo>(aiApiPath(`/videos/${video_id}`));
  }

/** Delete video */
  async deleteVideo(video_id: string | number): Promise<OpenAiVideoDeleteResponse> {
    return this.client.delete<OpenAiVideoDeleteResponse>(aiApiPath(`/videos/${video_id}`));
  }
}

export function createVideoApi(client: HttpClient): VideoApi {
  return new VideoApi(client);
}
